src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1789,6 +1805,37 @@ ReorderTaskPlacementsByTaskAssignmentPolicy(Job *job,
 }
 
 
+/*
+ * RemoveCoordinatorPlacement gets a task placement list and returns the list
+ * by removing the placement belonging to the coordinator (if any).
+ *
+ * If the list has a single entry or no placements on the coordinator, the list
+ * is return unmodified.
+ */
+static List *
+RemoveCoordinatorPlacement(List *placementList)
+{
+	ListCell *placementCell = NULL;
+
+	if (list_length(placementList) < 2)
+	{
+		return placementList;
+	}
+
+	foreach(placementCell, placementList)
+	{
+		ShardPlacement *placement = (ShardPlacement *) lfirst(placementCell);
+
+		if (placement->groupId == COORDINATOR_GROUP_ID)
+		{
+			return list_delete_ptr(placementList, placement);
```

Comment:
Minor: function signature feels like it doesn't modify its input, and returns a newly allocated list if needed. Unfortunately, `list_delete_ptr` modifies the list. So I'd either make the return type `void`, or make this function pure.